collect_app/src/test/java/org/odk/collect/android/http/CollectServerClientTest.java

```diff
@@ -0,0 +1,45 @@
+package org.odk.collect.android.http;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.odk.collect.android.http.mock.MockHttpClientConnection;
+import org.odk.collect.android.utilities.DocumentFetchResult;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertNull;
+import static junit.framework.Assert.assertTrue;
+
+@RunWith(JUnit4.class)
+public class CollectServerClientTest {
+
+    private TestableCollectServerClient collectServerClient;
+
+    @Before
+    public void setup() {
+        collectServerClient = new TestableCollectServerClient(new MockHttpClientConnection());
```

Comment:
This can be provided via the `TestComponent` instead of actually creating an instance of it here.  So you can move this to the [TestModule](https://github.com/opendatakit/collect/blob/master/collect_app/src/test/java/org/odk/collect/android/injection/TestModule.java)
and then follow the example in the [setup](https://github.com/opendatakit/collect/blob/master/collect_app/src/test/java/org/odk/collect/android/sms/SmsSenderJobTest.java#L44) method of a SMS test.

Reply:
Updated, added relevant @Provides to the TestModule and modified the setup() method